### ISSUE TITLE
[spring] Invalid default values for arrays and maps on API method declarations

### ISSUE THREAD

**Issue by rubms** [OpenAPITools/openapi-generator#1551]
##### Description

When parameters of type array are added to an operation, the generated `Spring` code includes an invalid `defaultValue` in the Spring MVC parameter annotations. For instance, an array of strings query parameter would generate a Spring code like the following:
```java
@Valid @RequestParam(value = "stringArrayParam", required = false, default="new ArrayList<>()") List<String> stringArrayParam
```

That generated `defaultValue` is invalid: the Spring MVC application will throw an error when a request is performed with no value specified for the parameter.

##### openapi-generator version

Reproduced with openapi-generator v3.3.3 and previous versions.

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0
info:
  description: "Spec for testing the default value of array parameters"
  version: 1.0.0
  title: Array parameters test case.
paths:
  /pets:
    get:
      operationId: getPets
      summary: Pets.
      description: "Get pets."
      parameters:
      - name: stringArrayParam
        in: query
        description: "Test array query parameter"
        required: false
        schema:
          type: array
          items:
            type: string
      tags:
      - Pets
      responses:
        200:
          description: Successful request.
components:
  schemas:
    ExampleObject:
      type: object
      properties:
        arrayProperty:
          type: array
          items:
            type: string
```

##### Command line used for generation
`java -jar openapi-generator-cli-3.3.3.jar generate -g spring -i test.yml -o test`

Additionally, the configuration property `library` is set to `spring-mvc`:
```
{
	"library": "spring-mvc"
}
```

##### Steps to reproduce

1. Generate a Spring-MVC server application from the previous specification. 
2. Run the application.
3. Throw a GET request to the `/pets` path (`getPets` operation) leaving the `stringArrayParam` query parameter blank or unspecified.

The server will throw the following error: 
```Failed to convert value of type 'java.lang.String' to required type 'java.util.List'; nested exception is org.springframework.core.convert.ConversionFailedException: Failed to convert from type [java.lang.String] to type [@io.swagger.annotations.ApiParam @javax.validation.Valid @org.springframework.web.bind.annotation.RequestParam com.rfranco.ics.inputdataapi.autogenerated.dtos.GameType] for value 'new ArrayList<>()'; nested exception is java.lang.IllegalArgumentException: Unexpected value 'new ArrayList<>()'
```

##### Related issues/PRs

This is related to #890, though not the same issue.

##### Suggest a fix/enhancement

The problem is that there are 2 different cases with respect to default values:

* The default value specified by the user, that should be included as default value in the Spring MVC API method annotations.
* The default variable initializer, which may be affected by the default value specified by the user but differs in data-types like arrays or maps.

However the Spring code generator uses the same value (the default variable initializer, in my opinion) for both: initializing variables and specifying Spring MVC annotations.

The Spring MVC code generator must use a different "default value" for both cases.

**Comment by wing328** [comment]
Closed via #1725

**Comment by mgvinuesa** [comment]
Hello, at least in version 4.3.1 this issue appears again, reviewing the source code I just check that the commit 47afef026c0ecefe8bbdbd6ef59c898e2b047761 has been reverted. I don´t know why or if there is another option to skip this problem.
